modcc/CMakeLists.txt

```diff
@@ -12,25 +16,24 @@ set(MODCC_SOURCES
     symdiff.cpp
     symge.cpp
     token.cpp
-
     io/prefixbuf.cpp
-
     printer/cexpr_emit.cpp
     printer/cprinter.cpp
     printer/cudaprinter.cpp
     printer/infoprinter.cpp
     printer/printerutil.cpp
 )
 
-include_directories("${PROJECT_SOURCE_DIR}/modcc")
+set(modcc_sources modcc.cpp)
 
-add_library(compiler ${MODCC_SOURCES})
+add_library(libmodcc STATIC ${libmodcc_sources})
+target_include_directories(libmodcc PUBLIC .)
+set_target_properties(libmodcc PROPERTIES OUTPUT_NAME modcc)
 
-add_executable(modcc modcc.cpp)
+add_executable(modcc ${modcc_sources})
+target_link_libraries(modcc libmodcc ext-tclap)
 
-target_link_libraries(modcc LINK_PUBLIC compiler)
+set_target_properties(modcc libmodcc PROPERTIES EXCLUDE_FROM_ALL ${ARB_WITH_EXTERNAL_MODCC})
+
+install(TARGETS modcc RUNTIME DESTINATION ${CMAKE_INSTALL_LIBEXECDIR})
```

Comment:
This is installed in `CMAKE_INSTALL_LIBEXECDIR`, which on my system is `<prefix>/libexe`.
`lmorpho` is installed in `CMAKE_INSTALL_BINDIR`, i.e. `<prefix>/bin`.
I would rather see modcc get installed in `bin`, what do you think?

Reply:
Put it in `libexec` on the grounds that it was a tool used to build the library, rather than a user executable, but on reflection this was wrong thinking.

If `modcc` were only ever used by `libarbor` — say to build mechanisms from NMODL at run time — then it would go here. But it is in fact a user-tool, and should go in `bin` as you suggest.